openerp/models.py

```diff
@@ -5809,15 +5811,16 @@ def onchange(self, values, field_name, field_onchange):
                             result['value'][name] = field.convert_to_write(
                                 newval, record._origin, subfields.get(name),
                             )
-                            todo.add(name)
+                            todo.append(name)
                         else:
                             # clean up result to not return another value
                             result['value'].pop(name, None)
 
         # At the moment, the client does not support updates on a *2many field
         # while this one is modified by the user.
-        if field_name and self._fields[field_name].type in ('one2many', 'many2many'):
-            result['value'].pop(field_name, None)
+        for name in names:
```

Comment:
Fixed as well, I added the condition `not isinstance(field_name, list)` to cover that case only.